src/main/java/com/gildedrose/AgedBrie.java

```diff
@@ -1,27 +1,29 @@
 package com.gildedrose;
 
-public class AgedBrie {
+public class AgedBrie implements Ageable {
     private int qualityIncrementation;
+    private final Item item;
 
-    public AgedBrie() {
+    public AgedBrie(Item item) {
+        this.item = item;
         this.qualityIncrementation = 1;
     }
 
-    public void age(Item item) {
-        item.quality = updateQuality(item.quality, item.sellIn);
-        item.sellIn = updateSellIn(item.sellIn);
+    public void age() {
+        item.quality = updateQuality();
+        item.sellIn = updateSellIn();
     }
 
-    private int updateQuality(int quality, int sellIn) {
-        if (sellIn <= 0) {
+    private int updateQuality() {
+        if (item.sellIn <= 0) {
             this.qualityIncrementation = 2;
         }
-        quality += this.qualityIncrementation;
-        return Math.min(quality, 50);
+        item.quality += this.qualityIncrementation;
```

Comment:
**Suggestion**: Possible to reduce the amount of if... else... statements w/ ternaries or something else more concise?

Reply:
Fwiw, I find if/else easier to read than ternaries.

Reply:
I would also agree that I find if/else's easier to read, so that would be my vote! What is everyone elses thoughts?

Reply:
Oh I didnt see your message @jramnani before I sent my comment on the same subject above

> This is a good question cause I am unsure about ternaries. They definitely make the code more concise but I feel like at a certain point they become more of a pain to read and more showing off that you can do some nifty hack. I'd appreciate input from anyone who wants to on this subject. I'm asking because I want to get ahead of this early in my career and hear different opinions on when to use them and when to avoid them.

Reply:
My vote here goes to `if... else` because of the logic in the conditional statement. If it was something like as simple as returning a value for example:
```java
(item.sellin <= 0 ) ? 5 : 4
```
then tenary is fine.

Reply:
Just please, if I teach you nothing else, do not nest ternaries. It makes reading code just awful.